infrastructure/src/main/java/org/corfudb/infrastructure/ResourceQuota.java

```diff
@@ -0,0 +1,39 @@
+package org.corfudb.infrastructure;
+
+import org.corfudb.runtime.exceptions.QuotaExceededException;
+
+/**
+ * A quota for a resource that can be modeled with a long.
+ *
+ * Created by Maithem on 6/20/19.
+ */
+public class ResourceQuota {
+
+    private final String name;
+    volatile long used;
+    private final long limit;
+
+    public ResourceQuota(String name, long limit) {
```

Comment:
limit in bytes? Maybe it's better to create A class like InformationUnit for that? Something like this https://github.com/JakeWharton/byteunits
similar to TimeUnit